### PR TITLE
feat(baremetalsolution): generate library

### LINKED ISSUE #8729: Baremetal Solution API: `baremetalsolution.googleapis.com`

### PR DESCRIPTION
**Pull request by dbolduc**
Fixes #8729

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8991)

### CODE REVIEW

File: external/googleapis/update_libraries.sh [hunk 1]
```diff
@@ -33,6 +33,7 @@ declare -A -r LIBRARIES=(
   ["artifactregistry"]="@com_google_googleapis//google/devtools/artifactregistry/v1:artifactregistry_cc_grpc"
   ["asset"]="@com_google_googleapis//google/cloud/asset/v1:asset_cc_grpc"
   ["assuredworkloads"]="@com_google_googleapis//google/cloud/assuredworkloads/v1:assuredworkloads_cc_grpc"
+  ["baremetalsolution"]="@com_google_googleapis//google/cloud/baremetalsolution/v2:baremetalsolution_cc_grpc"
```
Comment (coryan) [inline external/googleapis/update_libraries.sh:36]: nit: after `automl`

Comment (dbolduc) [reply external/googleapis/update_libraries.sh]: oops. fixed.

File: google/cloud/baremetalsolution/doc/main.dox [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+@include quickstart.cc
+
+## Environment Variables
+
+<!-- inject-endpoint-env-vars-start -->
+
+- `GOOGLE_CLOUD_CPP_BARE_METAL_SOLUTION_ENDPOINT=...` changes the default endpoint
+  (baremetalsolution.googleapis.com) used by `BareMetalSolutionConnection`.
+
+<!-- inject-endpoint-env-vars-end -->
```
Comment (devbww) [inline google/cloud/baremetalsolution/doc/main.dox:47]: Q. Should we be leaving the "inject-endpoint-env-vars" markers in the output?

Comment (dbolduc) [reply google/cloud/baremetalsolution/doc/main.dox:47]: yes, so the operation is idempotent

Comment (devbww) [reply google/cloud/baremetalsolution/doc/main.dox:47]: Then I guess we best go and add them to all the existing `main.dox` files.

**Pull request merged**